appengine/genetics/js/blocks.js

```diff
@@ -0,0 +1,445 @@
+      option.callback = Blockly.ContextMenu.callbackFactory(this, xmlBlock);
+      options.push(option);
+    }
+  }
+};
+
+/**
+ * Defines the JavaScript generation for acceptMate.
+ * @type {!Function}
+ */
+Blockly.JavaScript['genetics_acceptMate'] =
+    goog.partial(Blockly.JavaScript['genetics_generateMouseFunctionJS_'],
+                 'acceptMate', 'suitor');
+
+Blockly.Blocks['genetics_getSelf'] = {
+  /**
+   * Block for getting the mouse making the decision.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "%1(%2)",
+      "args0": ["getSelf", ""],
+      "output": "Mouse",
+      "colour": Genetics.Blocks.GENETICS_HUE,
+      "tooltip": BlocklyGames.getMsg('Genetics_getSelfTooltip')
+    });
+  }
+};
+
+/**
+ * Defines the JavaScript generation for getSelf.
+ * @param {Blockly.Block} block
+ * @return {!Array.<string|number>}
+ */
+Blockly.JavaScript['genetics_getSelf'] = function(block) {
+  // Generate JavaScript function that returns the mouse object making the
+  // decision.
+  return ['getSelf()', Blockly.JavaScript.ORDER_FUNCTION_CALL];
+};
+
+Blockly.Blocks['genetics_getMice'] = {
+  /**
+   * Block for getting all mice in the cage.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "%1(%2)",
+      "args0": ["getMice", ""],
+      "output": "Array",
+      "colour": Genetics.Blocks.GENETICS_HUE,
+      "tooltip": BlocklyGames.getMsg('Genetics_getMiceTooltip')
+    });
+  }
+};
+
+/**
+ * Defines the JavaScript generation for getMice.
+ * @param {Blockly.Block} block
+ * @return {!Array.<string|number>}
+ */
+Blockly.JavaScript['genetics_getMice'] = function(block) {
+  // Generate JavaScript for getting mice.
+  return ['getMice()', Blockly.JavaScript.ORDER_FUNCTION_CALL];
+};
+
+/**
+ * Block for getting the properties of a mouse.
+ */
+Blockly.Blocks['genetics_getProperties'] = {
+  /**
+   * Initialization of the getProperties block.
+   * @this Blockly.Block
+   */
+  init: function() {
+    var PROPERTIES =
+        [['size', 'SIZE'],
+         ['aggressiveness', 'AGGRESSIVENESS'],
+         ['startAggressiveness', 'START_AGGRESSIVENESS'],
+         ['fertility', 'FERTILITY'],
+         ['startFertility', 'START_FERTILITY'],
+         ['sex', 'SEX'],
+         ['age', 'AGE'],
+         ['id', 'ID'],
+         ['pickFightOwner', 'PICK_FIGHT'],
+         ['proposeMateOwner', 'PROPOSE_MATE'],
+         ['acceptMateOwner', 'ACCEPT_MATE']];
+    this.setColour(Genetics.Blocks.GENETICS_HUE);
+    this.appendValueInput('MOUSE')
+        .setCheck('Mouse');
+    var dropdown = new Blockly.FieldDropdown(PROPERTIES, function(newProp) {
+      thisBlock.updateType_(newProp);
+    });
+    this.appendDummyInput()
+        .appendField(dropdown, 'PROPERTY');
+    this.setInputsInline(true);
+    this.setOutput(true, 'Number');
+    // Assign 'this' to a variable for use in the tooltip closure below.
+    var thisBlock = this;
```

Comment:
It works, but I don't like that `thisBlock` is defined 7 lines after it is used in the dropdown closure.  Move this higher, and change the comment from "tooltip closure" to "closures".